quinn-proto/src/connection/mod.rs

```diff
@@ -70,8 +70,18 @@ where
     handshake_cid: ConnectionId,
     /// The CID the peer initially chose, for use during the handshake
     rem_handshake_cid: ConnectionId,
+    /// Cid rotation
+    ///
+    /// Expiration timestamp of local connection IDs
+    cids_timeline: VecDeque<CidTimeStamp>,
```

Comment:
Please expand on which CIDs this contains (all issued CIDs the peer hasn't retired? the highest sequence number of every batch of CIDs that share an expiration time, and that time? something else?). Consider renaming the field/type for clarity as well.

Reply:
Ack. When I will revisit this field definition now, i do feel it is lack of some explanation/comments. Will update in next revision.

Reply:
To give a brief explanation here,  `cids_timeline` records the cid sequence number (attached with a timestamp that shows when this cid is created) that were issued but not yet retired by peer.

Specifically, since quinn may issue multiple cids in one batch, for each record in `cids_timeline`, it shows the highest cid sequence number (in a batch) and the corresponding timestamp (since those cids in a batch are created at the same time).

Reply:
Correction: the timestamp in this struct is not when cid is created, but created + duration = when cid should be retired